src/components/UI/atoms/Icon/Icon.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import * as icomoon from 'styled-icons/icomoon';
+import { iconSizes, icons } from '<variables>/icons';
+import { iconColors } from '<variables>/colorPalette';
+import spacing from '<variables>/spacing';
+
+
+/**
+ * @description - Icon Component
+ *
+ * @prop {string} color - color
+ * @prop {string} height - height
+ * @prop {string} width - width
+ * @prop {string} display - display
+ * @prop {string} margin - margin
+ * @prop {string} icon - icon
+ *
+ * @return {component} Icon
+ */
+const Icon = ({
+  id,
+  color,
+  height,
+  width,
+  display,
+  margin,
+  icon,
+}) => {
+  const IconType = icomoon[icons[icon]];
+  return (
+    <Icon.Container
+      data-testid={id}
+      color={color}
+      height={height}
+      width={width}
+      display={display}
+      margin={margin}
+      icon={icon}
+    >
+      <IconType />
+    </Icon.Container>);
+};
+
+Icon.propTypes = {
+  children: PropTypes.node,
+  id: PropTypes.string,
+  color: PropTypes.oneOf(Object.keys(iconColors)),
+  icon: PropTypes.oneOf(Object.keys(icons)).isRequired,
+  height: PropTypes.oneOf(Object.keys(iconSizes)),
+  width: PropTypes.oneOf(Object.keys(iconSizes)),
+  display: PropTypes.oneOf(['block', 'inline', 'inline-block', 'none']),
+  margin: PropTypes.oneOf(Object.keys(spacing)),
+};
+
+Icon.defaultProps = {
+  id: 'mockIcon',
+  icon: 'search',
```

Comment:
Why would you want your default iconName be search

Reply:
The component requires that an iconName is supplied for it to render, as it is a required prop-type and search is one of the icons we need on the homepage so I decided to use it as default.